plugins/org.eclipse.xtext.generator/src/org/eclipse/xtext/generator/parser/antlr/splitting/internal/LexerSpecialStateTransitionSplitter.java

```diff
@@ -27,10 +27,11 @@
 			
 	public static final Pattern CASE_PATTERN = Pattern.compile(
 			"(^\\s*case\\s+(\\d+)\\s*:(\\s*))" +// case # -> $1, $2, $3
-			"([^;]*;)" + // int .. = input.LA(..) -> $4
+			"([^;]*;" + // int .. = input.LA(..) -> $4
```

Comment:
Here is an unclosed group in the regex. Could you please mention that in the comment after the line, e.g where this group is supposed to end.